rfc/composite_table_availability.pseudocode

```diff
@@ -0,0 +1,123 @@
+
+PhysicalTableSchema:
+	getFieldName(Column c)
+
+DataSource:
+	String getName()
+	Map<FieldName, Set<Segment>> raw segment data
+	Map<FieldName, List<Interval>> squashedFromSegments  // optional optimization
+
+	Set<FieldName> getFields() //union of all keys
+	List<Interval> getIntervals(FieldName fieldName)
+	Map<FieldName,List<Interval>> getIntervals(Set<FieldName>) // map slice from squashedMap
+	Set<SegmentId> getSegments(List<Interval>)  // Filter all segments by overlap on time
```

Comment:
This method here confuses me... The rest make sense, since they are really just helpers for getting at the internal `Map<FieldName, Set<Segment>>` in a clean, simple, non-map way, but this method is less clear as to why it's here.

I'm guessing it's here to serve the cache fingerprint use-case? If so, it's not clear to me how it will do it's job and be sure it's correct, since there's no top-level `Set<Segment>` structure in here, only one broken out by `FieldName`, which doesn't preclude having different segments for the same time ranges for different `FieldName`s?